imgui-SFML.cpp

```diff
@@ -553,7 +547,11 @@ void Image(const sf::Texture& texture,
 void Image(const sf::Texture& texture, const sf::Vector2f& size,
     const sf::Color& tintColor, const sf::Color& borderColor)
 {
-    ImGui::Image((void*)texture.getNativeHandle(), size, ImVec2(0, 0), ImVec2(1, 1), tintColor, borderColor);
+    GLuint nativeHandle = texture.getNativeHandle();
+    ImTextureID textureID;
+    std::memcpy(&textureID, &nativeHandle, sizeof(GLuint));
```

Comment:
These three lines are repeated in quite a few places now... Let's make some function: `ImTextureID convertToImGuiTextureHandle(GLuint)`

Plus, don't forget to initialize ImTextureID there (to `NULL`, I guess...), otherwise some bits can be garbage when `sizeof(GLUint) < sizeof(ImTextureId)`

Reply:
I've added the helper function - I don't believe the initialisation of `ImTextureID` is necessary as we never read from it directly, only ever memcpy (which is always valid) back into a `GLuint` which we then read from. Thus, even with garbage, I don't think this is UB (nor would it lead to inconsistent read/writes as we always use `sizeof(GLuint)`). I am hesitant to initialise it as when I checked on godbolt, initialising seemed to add an extra instruction on the latest gcc.

Reply:
On second thoughts, the copying of the ImTextureIDs around likely counts as a direct read so I'll go ahead and add the initialisation.